src/updatechecker.cpp

```diff
@@ -219,52 +220,80 @@ UpdateChecker::UpdateChecker(): Thread("WinSparkle updates check")
 
 void UpdateChecker::Run()
 {
-    // no initialization to do, so signal readiness immediately
-    SignalReady();
-
-    try
-    {
-        const std::string url = Settings::GetAppcastURL();
-        if ( url.empty() )
-            throw std::runtime_error("Appcast URL not specified.");
-        CheckForInsecureURL(url, "appcast feed");
-
-        StringDownloadSink appcast_xml;
-        DownloadFile(url, &appcast_xml, this, GetAppcastDownloadFlags());
-
-        Appcast appcast = Appcast::Load(appcast_xml.data);
-        if (!appcast.ReleaseNotesURL.empty())
-            CheckForInsecureURL(appcast.ReleaseNotesURL, "release notes");
-        if (!appcast.DownloadURL.empty())
-            CheckForInsecureURL(appcast.DownloadURL, "update file");
-
-        Settings::WriteConfigValue("LastCheckTime", time(NULL));
-
-        const std::string currentVersion =
-                WideToAnsi(Settings::GetAppBuildVersion());
-
-        // Check if our version is out of date.
-        if ( !appcast.IsValid() || CompareVersions(currentVersion, appcast.Version) >= 0 )
-        {
-            // The same or newer version is already installed.
-            UI::NotifyNoUpdates(ShouldAutomaticallyInstall());
-            return;
-        }
-
-        // Check if the user opted to ignore this particular version.
-        if ( ShouldSkipUpdate(appcast) )
-        {
-            UI::NotifyNoUpdates(ShouldAutomaticallyInstall());
-            return;
-        }
+	while (1)
+	{
+		// no initialization to do, so signal readiness immediately
+ 		SignalReady();
+
+		bool checkUpdates;
+		if (Settings::ReadConfigValue("CheckForUpdates", checkUpdates))
+		{
+			if (checkUpdates)
+			{
+				static const time_t ONE_DAY = 60 * 60 * 24;
```

Comment:
Agreed, I'm not using it and doesn't look like it was used before in the base Winsparkle library. I'll remove it.